Testing: Create UI test target and first test

rachelmcr opened the pull request:
This gets us started with UI tests with the following changes:

* Adds the `WooCommerceUITests` target with the first test (login/logout with email and password).
* Sets up test plans for unit tests and UI tests.
* Moves resources that are shared between the screenshot test and functional UI tests (Mocks, Screens, Utils) into the WooCommerceUITests target, for clarity in UI test development/maintenance.

To test:

* Confirm `UnitTests` is the default test plan (under the Product > Test Plan menu or in the Xcode test navigator) and all tests pass.
* Run `rake mocks` to start the mock server.
* Select the `UITests` test plan (under the Product > Test Plan menu or in the Xcode test navigator) and run the test in `WooCommerceUITests` to confirm it passes.
* Select the `WooCommerceScreenshots` scheme, run the screenshot test, and confirm it passes.

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

shiki commented on WooCommerce/WooCommerceUITests/Tests/LoginTests.swift:
```diff
@@ -0,0 +1,33 @@
+        .selectLogin()
+        .proceedWith(email: TestCredentials.emailAddress)
+        .proceedWithPassword()
+        .proceedWith(password: TestCredentials.password)
```
❤️ I really like this convention for UI tests

shiki commented on WooCommerce/WooCommerceUITests/Screens/Login/LoginEpilogueScreen.swift:
```diff
@@ -0,0 +1,38 @@
+    static let continueButton = "login-epilogue-continue-button"
+}
+
+class LoginEpilogueScreen: BaseScreen {
```
It's good practice to mark classes as `final` to be explicit that this class should never be inherited.

```suggestion
final class LoginEpilogueScreen: BaseScreen {
```

shiki commented on WooCommerce/WooCommerceUITests/Screens/Login/LoginEpilogueScreen.swift:
```diff
@@ -0,0 +1,38 @@
+class LoginEpilogueScreen: BaseScreen {
+    let continueButton: XCUIElement
+    let displayNameField: XCUIElement
+    let siteUrlField: XCUIElement
```
This comment is applicable to all the `Screen` classes in this PR.

I like that we _force_ call sites to use the functions like `continueWithSelectedSite()`. I believe we can even be more explicit by making these `private` too. 

Making these properties public could give future developers (i.e. me) a false signal that they can use these properties directly. Unless that is the intent?

```suggestion
    private let continueButton: XCUIElement
    private let displayNameField: XCUIElement
    private let siteUrlField: XCUIElement
```

rachelmcr replied:
Good point! At this point our UI tests are focused on functional testing of flows through the app (rather than e.g. exhaustively testing every element on a screen) so I think it makes sense to make these properties private, to enforce using the methods for each page object.

I could imagine making these properties public if we decide to diversify the kinds of UI tests we have, but it'd be better to make that an explicit change in how we do things.

rachelmcr commented on WooCommerce/Classes/Authentication/Epilogue/StoreTableViewCell.swift:
```diff
@@ -33,6 +33,7 @@ class StoreTableViewCell: UITableViewCell {
         }
         set {
             nameLabel?.text = newValue
+            nameLabel?.accessibilityIdentifier = "name-label"
```
Aha! I missed that this is how it should be done, thank you. :)

rachelmcr commented:
Thank you so much, @shiki! Your feedback was really helpful. I've addressed your comments (including a fix for the run destination error - that was caused by an incorrect deployment setting in the Build Settings for the UI test target) so this one's ready for another look.
